elements/lisk-bft/src/bft.ts

```diff
@@ -195,25 +193,25 @@ export class BFT extends EventEmitter {
 			blockHeader.asset.maxHeightPreviouslyForged >= blockHeader.height ||
 			(blockHeader.height - blockHeader.asset.maxHeightPreviouslyForged <=
 				heightThreshold &&
-				blockHeader.generatorPublicKey !==
-					maxHeightPreviouslyForgedBlock.generatorPublicKey)
+				!blockHeader.generatorPublicKey.equals(
+					maxHeightPreviouslyForgedBlock.generatorPublicKey,
+				))
 		) {
 			return false;
 		}
 
 		return true;
 	}
 
-	public get finalizedHeight(): number {
-		return this.finalityManager.finalizedHeight;
-	}
-
-	public get maxHeightPrevoted(): number {
-		return this.finalityManager.chainMaxHeightPrevoted;
+	public async getMaxHeightPrevoted(): Promise<number> {
+		const bftState = await this._chain.dataAccess.getConsensusState(
+			CONSENSUS_STATE_DELEGATE_LEDGER_KEY,
+		);
+		return this.finalityManager.getMaxHeightPrevoted(bftState);
```

Comment:
Related to earlier comment, we can move this `this._chain.dataAccess.getConsensusState` directly in the `finalityManager.getMaxHeightPrevoted`